docs/tutorials/FlowTutorial.md

```diff
@@ -0,0 +1,1630 @@
+openroad -gui
+```
+
+In `Tcl Commands` section of the GUI run the following command:
+
+```
+source init_floorplan2.tcl
+```
+
+View the resulting core utilization of 30 created following floorplan:
+
+![Relative Floorplan](./images/core_util.png)
+
+#### Defining Placement Density
+
+To learn on placement density strategies for `ibex` design, go to
+`OpenROAD-flow-scripts/flow`. Type:
+
+```
+openroad -gui
+```
+
+Enter following commands in `Tcl Commands` section of GUI
+
+```
+read_lef ./platforms/sky130hd/lef/sky130_fd_sc_hd.tlef
+read_lef ./platforms/sky130hd/lef/sky130_fd_sc_hd_merged.lef
+read_def ./results/sky130hd/ibex/base/3_place.def
+```
+![ibex placement density 60](./images/ibex_pl_60.png)
+
+Change `CORE_UTILIZATION` and `PLACE_DENSITY` for the `ibex` design
+`config.mk` as follows.
+
+```
+export CORE_UTILIZATION = 40
+export CORE_ASPECT_RATIO = 1
+export CORE_MARGIN = 2
+export PLACE_DENSITY = 0.50
+```
+
+Re-run the `ibex` design with below command:
+
+```
+make DESIGN_CONFIG=./designs/sky130hd/ibex/config.mk
+```
+
+View the `ibex` design placement density heat map as shown below:
+
+![ibex placement density 50](./images/ibex_pl_50.png)
+
+So from above GUI understood that change in `CORE_UTILIZATION` from 20
+to 40 and placement density default 0.60 to 0.50 changes standard cell
+placement became widely spread.
+
+### Chip Level IO Pad Placement
+
+In this section, user will generate an I/O pad ring for the coyote design
+using a Tcl script.
+
+ICeWall is a utility to place IO cells around the periphery of a design,
+and associate the IO cells with those present in the netlist of the
+design.
+
+For I/O pad placement using ICeWall refer readme file below:
+
+```
+less ../../src/pad/README.md
+```
+
+Refer to the built-in examples under
+`OpenROAD-flow-scripts/tools/OpenROAD/src/ICeWall/test` directory.
+
+Launch openroad GUI, Type
+
+```
+cd ../../src/padl/test/
+openroad -gui
+```
+
+Run `coyote_tc_sky130.tcl` to view ICeWall based IO pad placement.
+
+From the GUI Tcl command prompt run below script,
+
+```
+source coyote_tc_sky130.tcl
+```
+
+Tcl file reference.
+
+![Coyote IO pad script](./images/coyote_io_pad_script.png)
+
+View the resulting IO pad ring in GUI as shown below
+
+![coyote pad ring](./images/coyote_pad_ring.png)
+
+### Power Planning And Analysis
+
+In this section, user will use the design `black_parrot` to create a
+power grid and run power analysis.
+
+Pdngen is used for power planning. Refer to the following `README.md`.
+
+```
+less ../../src/pdn/README.md
+```
+
+Refer to the built-in examples under `../../src/pdn/test`
+directory.
+
+Launch openroad GUI,
+
+```
+cd ../../src/pdn/test
+openroad -gui
+```
+
+To run `coyote_tc_sky130.tcl` to view ICeWall based IO pad placement
+for `black_parrot` design type:
+
+```
+source soc_bsg_black_parrot_nangate45.tcl
+```
+
+Refer to the section of the power domain network strategy file below:
+
+![black_parrot PDN Strategy](./images/blackparrot_pdn.png)

```

Comment:
please consider using the tcl interface for PDN for this tutorial as it has improved error checking and is generally more consistent with the rest of the flow.

Reply:
Yes, we will do that